web/modules/custom/dpl_patron_menu/src/Plugin/Block/PatronMenuBlock.php

```diff
@@ -0,0 +1,142 @@
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Drupal config factory to get FBS and Publizon settings.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configuration = $configuration;
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The service container.
+   * @param mixed[] $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param int $plugin_definition
+   *   The plugin implementation definition.
+   *
+   * @return \Drupal\dpl_loans\Plugin\Block\LoanListBlock|static
+   *   Loan list block.
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return mixed[]
+   *   The app render array.
+   *
+   * @throws \JsonException
+   */
+  public function build(): array {
+    // Alternative to this menu array here this could be loaded from a drupal
+    // generated menu. A place for further improvements.
+    $menu = [
+      [
+        "name" => $this->t("Loans", ["context" => 'Patron menu']),
+        "link" => Url::fromRoute('dpl_loans.list', [], ['absolute' => TRUE])->toString(),
+        "dataId" => "0",
+      ],
+      [
+        "name" => $this->t("Reservations", ["context" => 'Patron menu']),
+        "link" => Url::fromRoute('dpl_reservation.list', [], ['absolute' => TRUE])->toString(),
+        "dataId" => "10",
+      ],
+      [
+        "name" => $this->t("My list", ["context" => 'Patron menu']),
+        "link" => Url::fromRoute('dpl_favorites_list.list', [], ['absolute' => TRUE])->toString(),
+        "dataId" => "20",
+      ],
+      [
+        "name" => $this->t("Fees & Replacement costs", ["context" => 'Patron menu']),
+        "link" => Url::fromRoute('dpl_fees.list', [], ['absolute' => TRUE])->toString(),
+        "dataId" => "30",
+      ],
+      [
+        "name" => $this->t("My account", ["context" => 'Patron menu']),
+        "link" => Url::fromRoute('dpl_dashboard.list', [], ['absolute' => TRUE])->toString(),
+        "dataId" => "40",
+      ],
+    ];
+
+    $data = [
+      // Config.
+      "threshold-config" => $this->configFactory->get('dpl_library_agency.general_settings')->get('threshold_config'),
+      "menu-navigation-data-config" => json_encode($menu, JSON_THROW_ON_ERROR),
+      // Added in THEME_preprocess_dpl_react_app__menu to use the theme defined
+      // icon.
+      "profile_svg" => '',
```

Comment:
This is missing, right?

Reply:
yeah, something has fallen between backend/frontend... we will fix it